src/server/frontend_xwayland/xwayland_connector.h

```diff
@@ -44,10 +41,10 @@ class XWaylandConnector : public Connector
         std::function<void(std::shared_ptr<scene::Session> const& session)> const& connect_handler) const override;
 
     auto socket_name() const -> optional_value<std::string> override;
+
 private:
-    bool enabled;
-    std::shared_ptr<XWaylandServer> xwayland_server;
-    std::unique_ptr<dispatch::ThreadedDispatcher> xserver_thread;
+    std::function<std::unique_ptr<XWaylandServer>()> const start_xwayland;
+    std::unique_ptr<XWaylandServer> xwayland_server;
```

Comment:
Perhaps this ought to be a `std::experimental::optional`?

Reply:
Why? That would require a definition of a copyable `XWaylandServer`.

Reply:
I mean an optional unique_ptr to indicate that it can be null

Reply:
I won't block on this, we can always do it later